### PR TITLE
[Quant] Add the operator of decomposed fake quant per channel

### PR DESCRIPTION
**Pull request by leslie-fang-intel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #121297

**Summary**
Add the operator of `quantized_decomposed.fake_quant_per_channel` and test the forward and backward of this op with comparing to ATen.

**Test Plan**
```
python -u -m pytest -s -v test_cpu_repro.py -k test_decomposed_fake_quant_per_channel
```

**Next Step**
Optimize the performance: from the generated code of forward and backward graph, the code didn't vectorize.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Comment by jerryzh168** [comment]
thanks! can you a perf test locally and report the perf number comparison with the current torch fake quant op as well?

**Review by jerryzh168 — APPROVED** [review]
just saw the last comment, it's fine to test perf in next PR

File: test/inductor/test_cpu_repro.py [hunk 1]
```diff
@@ -1730,6 +1730,72 @@ def fn(test_args):
...
+            )
+            return res
+
+        def qdq(input, scales, zero_points, axis, quant_min, quant_max):
+            res = torch.ops.quantized_decomposed.fake_quant_per_channel(
+                input, scales, zero_points, axis, quant_min, quant_max
+            )
+            return res
+
+        def test_eager_aten_fake_quant(
```
Comment (jerryzh168) [inline test/inductor/test_cpu_repro.py:1746]: nit: might be better to rename `test` to be more descriptive, e.g. `run`

Comment (leslie-fang-intel) [reply test/inductor/test_cpu_repro.py]: Thanks for the suggestion and changed the test name.

**Comment by leslie-fang-intel** [comment]
Here are the Inductor CPP backend generated code:

- Forward: https://gist.github.com/leslie-fang-intel/f6c4a8b3a98882acbfa465887539e9a0#file-fake_quant_per_channel_forward_generated_code-py
- Backward: https://gist.github.com/leslie-fang-intel/f6c4a8b3a98882acbfa465887539e9a0#file-fake_quant_per_channel_backward_generated_code-py

**Comment by leslie-fang-intel** [comment]
> thanks! can you a perf test locally and report the perf number comparison with the current torch fake quant op as well?

Although current Inductor generated code didn't vectorize, I write a [benchmark](https://gist.github.com/leslie-fang-intel/4714d20e331e70f81da4de3211b9d7ce) to test its current performance and here are the results:

- ATen fq time is: 1.687239408493042
- decomposed qdq time is: 1.2523536682128906
- compiled decomposed qdq time is: 1.3800063133239746

Hi @jerryzh168, could you help take a look of the benchmark script and do you think this preliminary result looks good to you?

**Review by jgong5 — APPROVED** [review]

**Pull request closed without merging**